driver/node.go

```diff
@@ -248,27 +258,15 @@ func (d *Driver) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpublish
 	})
 	ll.Info("node unpublish volume called")
 
-	vol, _, err := d.doClient.Storage.GetVolume(ctx, req.VolumeId)
-	if err != nil {
-		return nil, err
-	}
-
-	source := getDiskSource(vol.Name)
-	mounted, err := d.mounter.IsMounted(source, req.TargetPath)
-	if err != nil {
+	ll.Info("unmounting the target path")
+	err := d.mounter.Unmount(req.TargetPath)
+	// Invalid argument is returned by the `mount` command in this container
+	// image when a path is already umounted. We only return an error for
+	// anything else
+	if err != nil && !strings.Contains(err.Error(), "Invalid argument") {
```

Comment:
Mhmm this string comparison feels like it’s a little fragile, is there another way to determine the error case?

Reply:
Good call, as I said I'll use `IsMounted` that doesn't rely on the `Source` so it's a safe bet